src/Main.cpp

```diff
@@ -1,27 +1,62 @@
 #include <iostream>
 #include <list>
 #include <memory>
+#include <signal.h>
 #include <genset_whisperpower_ddc/VariableSpeed.hpp>
 #include <genset_whisperpower_ddc/VariableSpeedMaster.hpp>
 #include <genset_whisperpower_ddc/ControlCommand.hpp>
 
 using namespace std;
 using namespace genset_whisperpower_ddc;
 
+bool stop = false;
+
 void usage(ostream& stream) {
     stream << "usage: genset_whisperpower_ddc_ctl URI CMD\n"
            << "where:\n"
-           << "  URI is a iodrivers_base URI\n"
+           << "    URI is a iodrivers_base URI\n"
            << "\n"
            << "Available Commands\n"
-           << "  send_control_command CONTROL_COMMAND: send a control command to the DDC GenVerter power controller\n"
-           << "    CONTROL_COMMAND can be start, stop or keep_alive\n"
-           << "  read_frame: read a frame and print its content\n"
+           << "    run: send start control command to the DDC GenVerter power controller and then keep it running by "
+           << "repeatedly sending the keep_alive command until the user press any key. When this happens, send the stop command to stop de generator.\n"
+           << "    read_frame: read a frame and print its content\n"
            << endl;
 }
 
-int main(int argc, char** argv)
-{
+void handler(int s) {
+    stop = true;
+}
+
+void getValidFrame(std::unique_ptr<VariableSpeedMaster> &master) {
```

Comment:
This is really "waitForValidFrame". You don't get it, you never return it.

Now, since the read_frame command needs the frame, instead of having 90% of this logic copied, just return the frame and (1) ignore it in 'run' and (2) use it in `read_frame`